packages/stargate/src/stargateclient.ts

```diff
@@ -268,7 +270,34 @@ export class StargateClient {
     if (this.tmClient) this.tmClient.disconnect();
   }
 
-  public async broadcastTx(tx: Uint8Array): Promise<BroadcastTxResponse> {
+  public async broadcastTx(
+    tx: Uint8Array,
+    timeoutMs = 60_000,
+    pollIntervalMs = 3_000,
+  ): Promise<BroadcastTxResponse> {
+    return new Promise((resolve, reject) => {
+      const rejectTimeout = setTimeout(reject, timeoutMs, `timeout of ${timeoutMs}ms exceeded`);
+      this.broadcastTxCommit(tx)
+        .catch((error) => {
+          try {
+            const errorJson = JSON.parse(error.message);
+            if (errorJson.code !== -32603) {
```

Comment:
Compare with the check here: https://github.com/tendermint/vue/blob/develop/packages/vuex/src/modules/common/relayers/lib/starportSigningClient.js#L21-L24
I'm not sure if that code could be from other causes, but matching the string seems undesirable if there's a nice numerical code to use.

Reply:
error code: -32603 is "Internal Error" and CAN be from causes other than the timeout AFAIK.

Hence the check on both in my original implementation.